### DISCUSSION TITLE
Data fields missing in `session.user`

### DISCUSSION THREAD

**Discussion by thefakequake** [nextauthjs/next-auth#2078] [Help] [answered]
I created a custom database adapter for MongoDB. My `createUser` function  is as so:
```js
async createUser(profile) {
  const user = stripUndefined({
    name: profile.name,
    email: profile.email,
    image: profile.image,
    emailVerified: profile.emailVerified ?? null,
    id: await genUserId()
  })
  await db.collection("Users").insertOne(user)
  return user
},
```
However, whenever I use `useSesssion()` in one of my pages and I do `session.user`, I don't get the `id` field. How do I fix this?

**Comment by balazsorban44** [comment, marked as answer]
https://next-auth.js.org/configuration/callbacks#session-callback

**Reply by thefakequake** [reply]
> https://next-auth.js.org/configuration/callbacks#session-callback

Could you please elaborate on how this solves my issue? Like, give a code example?

**Reply by thefakequake** [reply]
Never mind, I figured it out.
